fsliolib/CMakeLists.txt

```diff
@@ -1,36 +1,43 @@
 #File Generated by Hans J. Johnson
 #Please contact hans-johnson@uiowa.edu for making enhancments/corrections
+include(FetchContent)
+FetchContent_Declare(
+  niftiio
+  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../niftilib
+)
+FetchContent_GetProperties(niftiio)
+if(NOT niftiio_POPULATED)
+  FetchContent_Populate(niftiio)
+  add_subdirectory(${niftiio_SOURCE_DIR} ${niftiio_BINARY_DIR})
+endif()
 
-include_directories(${NIFTILIB_SOURCE_DIR} ${ZNZLIB_SOURCE_DIR})
-set(FSLIOLIB_SRC fslio.c)
 
 # Michael Hanke 2004-04-25:
 # Restructure the file to match those of niftilib and znzlib
-set(NIFTI_FSLIOLIB_NAME ${PACKAGE_PREFIX}fslio)
-
-add_library(${NIFTI_FSLIOLIB_NAME} ${FSLIOLIB_SRC} )
-target_link_libraries( ${NIFTI_FSLIOLIB_NAME} ${PACKAGE_PREFIX}niftiio)
-target_include_directories(${NIFTI_FSLIOLIB_NAME} PUBLIC
+set(NIFTI_LIB_NAME ${PACKAGE_PREFIX}fslio)
+add_library(${NIFTI_LIB_NAME} fslio.c)
+target_include_directories(${NIFTI_LIB_NAME} PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-        $<INSTALL_INTERFACE:${NIFTI_INSTALL_INCLUDE_DIR}>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
         )
 
 # Set lib version when buildung shared libs.
 if(BUILD_SHARED_LIBS)
-  set_target_properties(${NIFTI_FSLIOLIB_NAME} PROPERTIES ${NIFTI_LIBRARY_PROPERTIES})
+  set_target_properties(${NIFTI_LIB_NAME} PROPERTIES ${NIFTI_LIBRARY_PROPERTIES})
 endif()
 
+target_link_libraries( ${NIFTI_LIB_NAME} ${PACKAGE_PREFIX}niftiio)
 if(NOT NIFTI_INSTALL_NO_LIBRARIES)
-  install(TARGETS ${NIFTI_FSLIOLIB_NAME}
+  install(TARGETS ${NIFTI_LIB_NAME}
           EXPORT NIFTITargets
-    RUNTIME DESTINATION ${NIFTI_INSTALL_BIN_DIR} COMPONENT RuntimeLibraries
-    LIBRARY DESTINATION ${NIFTI_INSTALL_LIB_DIR} COMPONENT RuntimeLibraries
-    ARCHIVE DESTINATION ${NIFTI_INSTALL_LIB_DIR} COMPONENT Development)
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT RuntimeLibraries
```

Comment:
This will likely break integration with ITK.  I'll need more time to review.

Reply:
Sure that's fine. If the gnuinstalldirs module doesn't have an equivalently convenient analog on mac and windows then its not worth it and abandoning this change for the original NIFTI_INSTALL...DIR variables is better.